cylc/flow/config_schema.py

```diff
@@ -0,0 +1,609 @@
+            'copyable environment variables': [VDR.V_STRING, ''],
+            'retrieve job logs': [VDR.V_STRING, ''],
+            'retrieve job logs command': [VDR.V_STRING, ''],
+            'retrieve job logs max size': [VDR.V_STRING, ''],
+            'retrieve job logs retry delays': [VDR.V_STRING, ''],
+            'task event handler retry delays': [VDR.V_STRING, ''],
+            'tail command template': [VDR.V_STRING, ''],
+            'owner': [VDR.V_STRING_LIST],
+            'batch systems': {
+                'err tailer': [VDR.V_STRING, ''],
+                'out tailer': [VDR.V_STRING, ''],
+                'err viewer': [VDR.V_STRING, ''],
+                'out viewer': [VDR.V_STRING, ''],
+                'job name length maximum': [VDR.V_STRING, ''],
+                'execution time limit': [VDR.V_INTERVAL_LIST],
+                'execution polling intervals': [VDR.V_INTERVAL_LIST],
+                'execution retry delays': [VDR.V_INTERVAL_LIST],
+                'batch submit command template': [VDR.V_STRING]
+            },
+            'default directives': {
+                '__MANY__': [VDR.V_STRING],
+            },
+        },
+    },
+    'email': {
+        'mail events': [VDR.V_STRING_LIST, None],
+        'mail from': [VDR.V_STRING],
+        'mail smtp': [VDR.V_STRING],
+        'mail to': [VDR.V_STRING],
+        'mail footer': [VDR.V_STRING],
+        'monitor': {
+            'sort order': [VDR.V_STRING, 'definition', 'alphanumeric'],
+        },
+    },
+    'scheduling': {
+        'initial cycle point': [VDR.V_CYCLE_POINT],
+        'final cycle point': [VDR.V_STRING],
+        'initial cycle point constraints': [VDR.V_STRING_LIST],
+        'final cycle point constraints': [VDR.V_STRING_LIST],
+        'hold after point': [VDR.V_CYCLE_POINT],
+        'cycling mode': (
+            [VDR.V_STRING, Calendar.MODE_GREGORIAN] +
+            list(Calendar.MODES) + ["integer"]),
+        'runahead limit': [VDR.V_STRING],
+        'max active cycle points': [VDR.V_INTEGER, 3],
+        'spawn to max active cycle points': [VDR.V_BOOLEAN],
+        'queues': {
+            'default': {
+                'limit': [VDR.V_INTEGER, 0],
+                'members': [VDR.V_STRING_LIST],
+            },
+            '__MANY__': {
+                'limit': [VDR.V_INTEGER, 0],
+                'members': [VDR.V_STRING_LIST],
+            },
+        },
+        'special tasks': {
+            'clock-trigger': [VDR.V_STRING_LIST],
+            'external-trigger': [VDR.V_STRING_LIST],
+            'clock-expire': [VDR.V_STRING_LIST],
+            'sequential': [VDR.V_STRING_LIST],
+            'exclude at start-up': [VDR.V_STRING_LIST],
+            'include at start-up': [VDR.V_STRING_LIST],
+        },
+        'xtriggers': {
+            '__MANY__': [VDR.V_XTRIGGER],
+        },
+        'graph': {
+            '__MANY__': [VDR.V_STRING],
+        },
+    },
+    'runtime': {
```

Comment:
I know `runtime` dates back forever, but it is ambiguous (server program runtime, or task jobs?). Can we change it to something like `task runtime`, or `job configuration`, or `jobs`?

Reply:
(probably not `jobs`)

Reply:
I like this idea, but again this is one for further comment from @oliver-sanders  and @dpmatthews. Other possibilities we might consider (just throwing ideas about):
* `tasks`
* `tasks runtimes`
* `task configurations`

Reply:
Is it really worth the user pain to change this?

Reply:
Only if we can agree a definitively better term?

Reply:
> Is it really worth the user pain to change this?

Yes!  Changing a single item name by deprecation (perhaps supported longer-term because it's a big one) is **not much pain**.  In light of recent discussions about how cylc is perceived by outsiders, we need to be getting rid of anything that looks confusing or quirky.  I took a step back from what I'm used to seeing every day, and though "what in hell does 'runtime' mean" - there is nothing in that word that suggests it is a job configuration section, as opposed to server program, or even validation etc. (it all involves a "runtime").

Reply:
#3404